Cargo.toml

```diff
@@ -17,6 +17,7 @@ links = "python"
 [dependencies]
 cfg-if = { version = "1.0" }
 # must stay at 0.3.x for Rust 1.41 compatibility
+eyre = {version = ">= 0.4" , optional = true}
```

Comment:
Think these need to be swapped (new dep has split comment from its target). Also, should we place an upper bound?

```suggestion
eyre = {version = ">= 0.4, <0.5" , optional = true}
# must stay at 0.3.x for Rust 1.41 compatibility
```

Reply:
(oops)

Eyre is at 0.6.5 now, so that upper bound is too strict. 0.4 is the first version that has the eyre::Report type (?), so I think any version above that should be fine.

I'll admit that I don't know anything about how cargo resolves dependency versions, but this looks like how you should say "works with any eyre version of 0.4 and up". Is that too liberal and should it be pinned to `">= 0.4, <0.7"` instead?

Reply:
I think that by default `cargo` will pick the highest `eyre` possible for the pyo3 dependency, not the one which the user already has in their own Cargo.toml. (This is a bit awkward.) The user can adjust it manually using some incantation of `cargo update --precise` (like we do in our MSRV CI jobs).

> Is that too liberal and should it be pinned to `">= 0.4, <0.7"` instead?

Generally we've only offered a range for dependencies when the MSRV support requires an older version. I'd be tempted to ask the other question - is just supporting `eyre = "0.6"` acceptable? 

As for the upper bound: if we were to take semver literally, eyre `0.7` could hypothetically be very breaking and completely remove the `eyre::Report` type. (Although I would be suprised if this happened in practice.) So I'd definitely prefer have an upper bound of `< 0.7` if we go for a range. Dependabot can remind us to update when 0.7 releases 😄